volatility3/framework/symbols/windows/extensions/__init__.py

```diff
@@ -761,6 +762,7 @@ def add_process_layer(
             self._context, dtb, config_prefix, preferred_name
         )
 
+    @functools.lru_cache
```

Comment:
Any reason to cache `get_peb` and `get_peb32` as well as add_process_layer?  Surely caching add_process_layer gives you the same benefits?

Reply:
I figured it saves a little bit of time reconstructing the objects, but that's probably negligible since I don't think either of those methods gets called in a hot loop.